controllers/bucket_controller.go

```diff
@@ -211,7 +213,7 @@ func (r *BucketReconciler) reconcile(ctx context.Context, bucket sourcev1.Bucket
 
 	// return early on unchanged revision
 	artifact := r.Storage.NewArtifactFor(bucket.Kind, bucket.GetObjectMeta(), revision, fmt.Sprintf("%s.tar.gz", revision))
-	if meta.HasReadyCondition(bucket.Status.Conditions) && bucket.GetArtifact().HasRevision(artifact.Revision) {
+	if meta.InReadyCondition(bucket.Status.Conditions) && bucket.GetArtifact().HasRevision(artifact.Revision) {
```

Comment:
Given the subtle difference, I wonder if it would be better to do

```suggestion
	if apimeta.IsStatusConditionTrue(bucket.Status.Conditions, meta.ReadyCondition) && bucket.GetArtifact().HasRevision(artifact.Revision) {
```

to get rid of another "custom" way of doing things.

Reply:
The pkg/apis `meta.InReadyCondition` actually abstracts the same `apimeta.IsStatusConditionTrue`, so I wouldn't necessarily say it's a custom way of doing things, the func is also part of the same package where `meta.ReadyCondition` is defined so they are coupled in that sense.

Do you think we should just get rid of `meta.InReadyCondition` altogether?

Reply:
Now that I think about it, and how little use it has, I suppose it's better to just ditch it. Will update my branch and submit a PR to remove `InReadyCondition` from pkg/apis/meta as well.

Reply:
https://github.com/fluxcd/pkg/pull/50